src/olympia/bandwagon/views.py

```diff
@@ -376,12 +376,12 @@ def add(request):
 def ajax_new(request):
     form = forms.CollectionForm(
         request.POST or None,
-        initial={'author': request.user, 'application': request.APP.id},
-    )
+        initial=initial_data_from_request(request))
 
     if request.method == 'POST' and form.is_valid():
         collection = form.save()
         addon_id = request.POST['addon_id']
```

Comment:
It's a bit annoying that this is not validated by the form - it will result in a 500 error if the `addon_id` or missing. But at least now  the hack is gone.